app/src/main/java/org/dhis2/usescases/main/MainPresenter.kt

```diff
@@ -0,0 +1,126 @@
+package org.dhis2.usescases.main
+
+import android.content.Context
+import android.text.TextUtils.isEmpty
+import android.view.Gravity
+import androidx.work.WorkManager
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.functions.Consumer
+import org.dhis2.data.prefs.Preference
+import org.dhis2.data.schedulers.SchedulerProvider
+import org.dhis2.usescases.login.LoginActivity
+import org.dhis2.utils.Constants
+import org.dhis2.utils.filters.FilterManager
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.user.User
+import timber.log.Timber
+
+const val DEFAULT = "default"
+
+class MainPresenter(
+        private val view: MainView,
+        private val d2: D2,
+        private val schedulerProvider: SchedulerProvider) {
+
+    private var compositeDisposable: CompositeDisposable? = null
+
+    fun init() {
+        this.compositeDisposable = CompositeDisposable()
```

Comment:
Should we initialize the disposable outside of the init method?

Reply:
Ye